test/utils/mock-provider.js

```diff
@@ -20,4 +20,8 @@ MockProvider.prototype.send = function (payload, callback) {
   }, this._delay)
 }
 
+MockProvider.prototype.on = function() {
+  console.warn('Suscriptions mocked but not implemented.')
+}
```

Comment:
`web3` was complaining that the provider was missing this `on` method.

Reply:
@gndelia what if these comments are added to the code instead of in the PR? Thinking that in the future anyone can then go to the code and understand the rationale. Otherwise that will be lost in GitHub and PRs.

Reply:
you're right - added here https://github.com/autonomoussoftware/metronome-wallet-core/pull/34/commits/460c2d783ae7e660aadf706d09642475b82c3cc6